### PR TITLE
Note the Magento Cart2Quote extension RCE vulnerability

### PR DESCRIPTION
**Pull request by andrewhowdencom**

### CODE REVIEW

**Comment by andrewhowdencom** [comment]
Either accept it or not, I don't mind (: It just came up; looking for a place to log Magento extension vulns.

File: magento/cart2quote/2017-02-01.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+title:     Remote Code Execution in Qquoteadv/controllers/DownloadController.php
+link:      https://cart2quote.zendesk.com/hc/en-us/articles/115000616303--FIXED-Security-Vulnerability-in-downloadCustomOptionAction
+cve:       ~
+branches:
+    4:
+        time:     2017-02-01 10:45:00
+        versions: ['>=4.1.6', '<5.0'
```
Comment (xabbuh) [inline magento/cart2quote/2017-02-01.yaml]: This is missing the closing square bracket.

File: magento/cart2quote/2017-02-01.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (xabbuh) [inline magento/cart2quote/2017-02-01.yaml]: not needed

File: magento/cart2quote/2017-02-01.yaml (hunk 1 above)
Comment (xabbuh) [inline magento/cart2quote/2017-02-01.yaml]: The advisory is by the way talking about 4.2.0 and above. Not sure what is right.

File: magento/cart2quote/2017-02-01.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+link:      https://cart2quote.zendesk.com/hc/en-us/articles/115000616303--FIXED-Security-Vulnerability-in-downloadCustomOptionAction
+cve:       ~
+branches:
+    4:
+        time:     2017-02-01 10:45:00
+        versions: ['>=4.1.6', '<5.0'
+    5:
+        time:     2017-02-01 10:45:00
+        versions: ['>=5.0.1', '<5.4.4']
+reference: composer://cart2quote/cart2quote
```
Comment (stof) [inline magento/cart2quote/2017-02-01.yaml]: this is not consistent with the location of the file

Comment (stof) [inline magento/cart2quote/2017-02-01.yaml]: according to their installation doc, the package is name ``cart2quote/module-quotation`` or ``cart2quote/module-quotation-encoded`` btw (depending on the version being used).
Using the wrong package name here would make the checker tool unable to detect the usage of a vulnerable version

File: magento/cart2quote/2017-02-01.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+title:     Remote Code Execution in Qquoteadv/controllers/DownloadController.php
+link:      https://cart2quote.zendesk.com/hc/en-us/articles/115000616303--FIXED-Security-Vulnerability-in-downloadCustomOptionAction
+cve:       ~
+branches:
+    4:
+        time:     2017-02-01 10:45:00
+        versions: ['>=4.1.6', '<5.0'
+    5:
+        time:     2017-02-01 10:45:00
+        versions: ['>=5.0.1', '<5.4.4']
```
Comment (stof) [inline magento/cart2quote/2017-02-01.yaml]: is 5.0.0 really safe ?

File: magento/cart2quote/2017-02-01.yaml (hunk 1 above)
Comment (stof) [inline magento/cart2quote/2017-02-01.yaml]: the page you link above has patched versions for all brances, not only for 5.4.4. So the list of affected versions is wrong

**Comment by andrewhowdencom** [comment]
Can I just say, I hadn't anticipated this would get attention let alone the super helpful comments that you've put on it. 10/10 would report again; I will amend those notes and fix up the PR. Will squash the commit so it's tidier.

File: magento/cart2quote/2017-02-01.yaml (hunk 3 above)
Comment (stof) [inline magento/cart2quote/2017-02-01.yaml]: if they have 2 package names, you will have to create 2 advisory files btw (one for each of them, even if everything else stays the same)

**Comment by stof** [comment]
@andrewhowdencom FYI, you can run ``php validator.php`` locally to ensure that the structure of your advisory is correct (instead of doing back-and-forth waiting for the CI to run each time you try to fix it). Some of the issues we reported are also detected by the validator (not all of them, as it cannot ensure that the content of the advisory matches the needed one of course)

File: magento/cart2quote/2017-02-01.yaml (hunk 1 above)
Comment (andrewhowdencom) [inline magento/cart2quote/2017-02-01.yaml]: How would I best represent this? 
```
    4.1.6:
        time:     2017-02-01 10:45:00
        versions: ['>=4.1.6', '<4.1.6-patch1']
```

Note: Still getting my head around the docs. If this is a foolish question, I'll find the answer shortly.

Comment (andrewhowdencom) [inline magento/cart2quote/2017-02-01.yaml]: There is a patch (well, a file) for 4.1.6, which changes the unserialize to json_decode, just as in 4.2.0. I'd suggest they missed the wording.

File: magento/cart2quote/2017-02-01.yaml (hunk 4 above)
Comment (andrewhowdencom) [inline magento/cart2quote/2017-02-01.yaml]: According to the release notes, there is no version 5.0.0.

Ref: https://www.cart2quote.com/release-notes#r501

File: magento/cart2quote/2017-02-01.yaml (hunk 1 above)
Comment (andrewhowdencom) [inline magento/cart2quote/2017-02-01.yaml]: Given that I can't find a logical way to represent patch versions, I'm marking all versions between 4.1.6 and 5.4.4 dodgey.

File: magento/cart2quote/2017-02-01.yaml (hunk 3 above)
Comment (andrewhowdencom) [inline magento/cart2quote/2017-02-01.yaml]: It looks like composer is used for the Magento 2 installation, which is not affected by this issue. I'm not sure how to best represent this. 

https://cart2quote.zendesk.com/hc/en-us/articles/115000616303

> All Magento 1 version

https://www.cart2quote.com/documentation/magento-2-cart2quote-installation-manual/

(See the title)

**Comment by andrewhowdencom** [comment]
PR Updated to the best of my current knowledge.

File: magento/cart2quote/2017-02-01.yaml (hunk 4 above)
Comment (stof) [inline magento/cart2quote/2017-02-01.yaml]: well, I would still say ``>=5.0``, to be clear

File: magento/cart2quote/2017-02-01.yaml (hunk 3 above)
Comment (stof) [inline magento/cart2quote/2017-02-01.yaml]: oh, the issue is only for magento 1 ?

File: magento/cart2quote/2017-02-01.yaml (hunk 4 above)
Comment (andrewhowdencom) [inline magento/cart2quote/2017-02-01.yaml]: Updated

**Comment by andrewhowdencom** [comment]
Is there anything else I can do to make this PR better? I'm looking to establish precedent for submitting Magento extensions; I'll create new PRs as I come across other extension (or core) vulns -- I also want to encourage others in the community to do so also.

**Comment by fabpot** [comment]
Let's merge! Thanks for your contribution.

**Comment by andrewhowdencom** [comment]
Thanks man! I appreciate it. There's such a need for a project such as this, and this one is well structured and already provides the tooling!

**Pull request merged**